gnat2goto/driver/ureal_to_binary.adb

```diff
@@ -0,0 +1,228 @@
+with Uintp; use Uintp;
+with Types; use Types;
+with Ada.Text_IO;
+
+package body Ureal_To_Binary is
+   package IO renames Ada.Text_IO;
+
+   procedure Get_Integer_And_Fractional_Parts (Number : Ureal;
+                                              Int_Part : out String;
+                                              Frac_Part : out String;
+                                              Int_End : out Integer;
+                                              Frac_End : out Integer)
+     with Pre => (Denominator (Number) > Uint_0);
+
+   procedure Get_Integer_And_Fractional_Parts (Number : Ureal;
+                                               Int_Part : out String;
+                                               Frac_Part : out String;
+                                               Int_End : out Integer;
+                                               Frac_End : out Integer) is
```

Comment:
Why are there so many out variables?

Reply:
Couldn't figure out a better way to return them. `String` is an indeterminate type, i.e. one that doesn't have a specific size, and thus can't be put into records (the actual definition is `type String is array(Integer range <>) of Character` IIRC), so I can't just return a record, and I wanted to return two strings (one for integer part and one for fractional) to make it easier to work with in the following functions. The `*_End` parameters could be returned, I just kept them as out parameters for symmetry (they give the index of the last meaningful character in the string).